src/plugins/showHiddenChannels.tsx

```diff
@@ -84,18 +105,33 @@ export default definePlugin({
             }
         }
     ],
+    shouldShow(channel, category, isMuted) {
+        if (!this.isHiddenChannel(channel)) return false;
+        if (!category) return false;
+        if (channel.type === 0 && category.guild?.hideMutedChannels && isMuted) return false;
+
+        return !category.isCollapsed;
+    },
     isHiddenChannel(channel) {
         if (!channel) return false;
-        channel._isHiddenChannel = !can(VIEW_CHANNEL, channel);
+        if (channel.channelId)
+            channel = ChannelStore.getChannel(channel.channelId);
+        if (!channel || channel.isDM() || channel.isGroupDM() || channel.isMultiUserDM())
+            return false;
+
+        // check for disallowed voice channels too so that they get hidden when collapsing the category
+        channel._isHiddenChannel = !can(VIEW_CHANNEL, channel) || (channel.type === 2 && !can(CONNECT, channel));
         return channel._isHiddenChannel;
     },
     channelSelected(channelData) {
+        if (!channelData) return false;
         const channel = ChannelStore.getChannel(channelData.channelId);
 
```

Comment:
```suggestion
        if (!channel) return false;
```
discord crashes on startup for me without this
```
66716e85bf16d7f6a198.js:6052 TypeError: Cannot read property 'type' of undefined
    at Object.channelSelected (VencordRenderer:14)
    at vce (VM3171 WebpackModule583156:2)
    at i.handleHistoryChange (VM3171 WebpackModule583156:2)
    at e.t.addRouteChangeListener (66716e85bf16d7f6a198.js:4115)
    at o.i.componentDidMount (VM3171 WebpackModule583156:2)
    at Ss (WebpackModule364448:2)
    at Vu (WebpackModule364448:2)
    at t.unstable_runWithPriority (66716e85bf16d7f6a198.js:1663)
    at Ko (WebpackModule364448:2)
    at Bu (WebpackModule364448:2)
```